test/watcher.test.js

```diff
@@ -0,0 +1,140 @@
+const path = require('path');
+const ncc = global.coverage ? require("../src/index") : require("../");
+const fs = require('fs');
+
+// Based on the NodeWatchFileSystem class at:
+// - https://github.com/webpack/webpack/blob/master/lib/node/NodeWatchFileSystem.js
+// which in turn exposes:
+// - https://www.npmjs.com/package/watchpack
+
+class CustomWatchFileSystem {
+  constructor(watchStart, watchEnd) {
+    this.closed = false;
+    // paused allows the watchers to stay open for the next build
+    this.paused = false;
+    this.changeCallback = undefined;
+    this.watchStart = watchStart;
+    this.watchEnd = watchEnd;
+
+    // Webpack requires us to track this stuff
+    this.files = undefined;
+    this.dirs = undefined;
+    this.missing = undefined;
+    this.timestamps = new Map();
+
+    // this will be populated for us by ncc
+    this.inputFileSystem = undefined;
+  }
+  
+  triggerChanges (changed, removed) {
+    if (!this.paused) {
+      const newTime = +Date.now();
+      for (const file of changed)
+        this.timestamps.set(file, {
+          safeTime: newTime + 10,
+          accuracy: 10,
```

Comment:
What do `safeTime` and `accuracy` represent?

Reply:
I've only been able to infer these myself looking at the implementation of http://npmjs.org/package/watchpack, perhaps @sokra can clarify further.

But I believe:

* safeTime represents an upper bound of the actual mtime, given that the mtime on its own might not be enough to trigger a cache invalidation
* timestamp being the mtime
* accuracy being the milliseconds from the best known timestamp that the safeTime is set to